### PR TITLE
--group support in intelmqctl

### PR DESCRIPTION
**Pull request by e3rd**
We can now handle a group of bots. (Previously, we handled only specific bot or whole botnet.) Will be very useful in intelmq-manager

### CODE REVIEW

**Comment by robcza** [comment]
@e3rd I love it :)
Definitely better than my manual approach:

```
#start all stopped experts
intelmqctl status | grep stopped | sed -ne 's/intelmqctl: \([^ ]*\) .*/\1/p' | grep expert | while read line ; do intelmqctl start $line ; done

#stop all running collectors
intelmqctl status | grep running | sed -ne 's/intelmqctl: \([^ ]*\) .*/\1/p' | grep collector | while read line ; do intelmqctl stop $line ; done
```

**Comment by e3rd** [comment]
Thanks, nice to hear from you, @robcza ! :)

File: docs/User-Guide.md [hunk 1]
```diff
@@ -165,6 +165,8 @@ This configuration is used by each bot to load the source pipeline and destinati
 }
 ```
 
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (e3rd) [reply docs/User-Guide.md]: sorry, corrected

File: CHANGELOG.md [hunk 2]
```diff
@@ -97,7 +99,7 @@ CHANGELOG
 - added destination.urlpath and source.urlpath to harmonization.
 
 ### Requirements
-- Requests is no longer listed as dependency of the core. For depending bots the requirement is noted in their REQUIREMENTS.txt file
+- Requests is no longer a listed as dependency of the core. For depending bots the requirement is noted in their REQUIREMENTS.txt file
```
[the remark this answers is not in the document]
Comment (e3rd) [reply CHANGELOG.md]: i didn't intend to correct this, don't know what happened :D

File: intelmq/bin/intelmqctl.py [hunk 3]
```diff
@@ -510,26 +522,36 @@ def __init__(self, interactive: bool = False, return_type: str = "python", quiet
             parser_start = subparsers.add_parser('start', help='Start a bot or botnet')
             parser_start.add_argument('bot_id', nargs='?',
                                       choices=self.runtime_configuration.keys())
+            parser_start.add_argument('--group', help='Start a group of bots',
+                                     choices=BOT_GROUP.keys())
```
Comment (wagner-certat) [inline intelmq/bin/intelmqctl.py]: As None is also in `BOT_GROUP` this leads to the following output:
```
intelmqctl stop: error: argument --group: invalid choice: 'foobar' (choose from 'experts', 'outputs', 'parsers', 'collectors', None)
```
None is no valid input (and can't be an input)

File: intelmq/bin/intelmqctl.py [hunk 4]
```diff
@@ -445,7 +457,7 @@ def __init__(self, interactive: bool = False, return_type: str = "python", quiet
             subparsers = parser.add_subparsers(title='subcommands')
 
             parser_list = subparsers.add_parser('list', help='Listing bots or queues')
-            parser_list.add_argument('kind', choices=['bots', 'queues'])
+            parser_list.add_argument('kind', choices=['bots', 'queues', 'queues-and-status'])
```
Comment (wagner-certat) [inline intelmq/bin/intelmqctl.py]: Could you please add that to `bash-completion/intelmq.sh` line 40?

File: intelmq/bin/intelmqctl.py [hunk 5]
```diff
@@ -709,6 +747,10 @@ def list(self, kind=None):
             return self.list_queues()
         elif kind == 'bots':
             return self.list_bots()
+        elif kind == 'queues-and-status':
+            q = self.list_queues()
+            b = self.botnet_status()
+            return q[0] + b[0], [q[1], b[1]]
```
Comment (wagner-certat) [inline intelmq/bin/intelmqctl.py]: Summing the return values is not correct, as 1 and 1 should be 1. Using `max()` or the logical or `|` would work better.

File: intelmq/bin/intelmqctl.py (hunk 3 above)
Comment (e3rd) [reply intelmq/bin/intelmqctl.py]: agree

File: intelmq/bin/intelmqctl.py (hunk 5 above)
Comment (e3rd) [reply intelmq/bin/intelmqctl.py]: Thanks for the hint. (I did the same (`|`) in the `bot_restart` method that we merged few days ago.)

File: intelmq/bin/intelmqctl.py (hunk 4 above)
Comment (e3rd) [reply intelmq/bin/intelmqctl.py]: thanks for introducing me into that; I've also added some missing completions in there

**Pull request merged**